### ISSUE TITLE
Do not error out when a cookie contains unknown or obsolete attributes

### ISSUE THREAD

**Issue by bilal-fazlani** [softwaremill/sttp#415]
Sample cookie: 

```
Set-Cookie: WELCOME_STATE_CHECKER=Xh3JYGEWrY7X2CIyI8_xqJ4UkipYtamLS2YOU27h8vY; Version=1; Expires=Tue, 28-Jan-2020 17:13:10 GMT; Max-Age=300; Path=/; HttpOnly
```

results into error: 
```
Unknown cookie attribute: version=1
```

`Version` is part of [RFC2109](https://tools.ietf.org/html/rfc2109#section-4.2.2) & [RFC2965](https://tools.ietf.org/html/rfc2965?page=5#section-3.2)

 >Version=value
 >    REQUIRED.  The value of the Version attribute, a decimal integer,
      identifies the version of the state management specification to
      which the cookie conforms.  

Both, RFC2109 & RFC2965 have been made obsolete by RFC6265 which does not require "Version" attribute. After going through [section-5.2](https://tools.ietf.org/html/rfc6265#section-5.2) & [section-5.3](https://tools.ietf.org/html/rfc6265#section-5.3), I feel that this library should not error out when it encounters an unknown attribute. The algorithm in Section 5.2 does not discriminate among attributes. It simply reads all of them and provides instructions on how to treat some attributes.

A more compliant behaviour might be to simply make unknown attributes of cookie available as part of a generic data structure such as a Map or List. More than compliance, I think the main advantage of taking this approach would be - it would make applications more resilient to older and obsolete standards.

**Comment by bilal-fazlani** [comment]
If there is agreement,  I can work on a PR for this

**Comment by adamw** [comment]
Yes, I think we do need to add a `Map[String, String]` of attributes. A PR would be great! Though that would be for the https://github.com/softwaremill/sttp-model project :)

**Comment by bilal-fazlani** [comment]
Awesome! I already started working on it. Just with one minor difference. What I have on my local machine is a `Map[String, Option[String]]` and not `Map[String, String]`. 

```scala
case class CookieValueWithMeta private (
    value: String,
    expires: Option[Instant],
    maxAge: Option[Long],
    domain: Option[String],
    path: Option[String],
    secure: Boolean,
    httpOnly: Boolean,
    otherAttributes: Map[String, Option[String]]
)
```

Because similar to known attributes such as path, domain, expires, etc, the unknown attributes can also have optional values. 

Will raise a PR in that repo after writing some unit tests.

**Comment by adamw** [comment]
Right, `Option` makes sense :)

**Comment by bilal-fazlani** [comment]
Thank you for merging the PR  @adamw . Hopefully there will be a release soon? 
I can then get back to consuming my cookies :)

**Comment by adamw** [comment]
Yes, sonatype works again, so released as 1.0.0-RC6. Should be available in 15-30 mins
